app/models/landing_page/block/main_navigation.rb

```diff
@@ -5,9 +5,10 @@ class MainNavigation < Base
     def initialize(block_hash, landing_page)
       super
 
-      @title = data.fetch("title")
-      @title_link = data.fetch("title_link")
-      @links = data.fetch("links")
+      navigation_group = landing_page.navigation_groups[data["navigation_group_id"]]
+      @title = navigation_group["links"].select { |l| l["theme"] == "homepage" }.first["text"]
+      @title_link = navigation_group["links"].select { |l| l["theme"] == "homepage" }.first["href"]
+      @links = navigation_group["links"].reject { |l| l["theme"] == "homepage" }
```

Comment:
Could these be methods? It would make them easier to read.
It would also be good to add tests for them.

Could also add something to the commit message about what the "homepage" theme is, (have we seen it before?) and why it's being excluded.

Reply:
I've removed the concept of theme and the idea it was supposed to support (that the first link in the main navigation was somehow special and needed to be extracted at this point rather than in the partial)